pkgs/build-support/setup-hooks/make-binary-wrapper.sh

```diff
@@ -0,0 +1,198 @@
+# Generate a binary executable wrapper for wrapping an executable.
+# The binary is compiled from generated C-code using gcc.
+# makeBinaryWrapper EXECUTABLE OUT_PATH ARGS
+
+# ARGS:
+# --argv0       NAME    : set name of executed process to NAME
+#                         (otherwise it’s called …-wrapped)
+# --set         VAR VAL : add VAR with value VAL to the executable’s
+#                         environment
+# --set-default VAR VAL : like --set, but only adds VAR if not already set in
+#                         the environment
+# --unset       VAR     : remove VAR from the environment
+# --add-flags   FLAGS   : add FLAGS to invocation of executable
+
+# --prefix          ENV SEP VAL   : suffix/prefix ENV with VAL, separated by SEP
+# --suffix
+
+# To troubleshoot a binary wrapper after you compiled it,
+# use the `strings` command or open the binary file in a text editor.
+makeBinaryWrapper() {
+    makeDocumentedCWrapper "$1" "${@:3}" | gcc -Os -x c -o "$2" -
+}
+
+# Generate source code for the wrapper in such a way that the wrapper source code
+# will still be readable even after compilation
+# makeDocumentedCWrapper EXECUTABLE ARGS
+# ARGS: same as makeBinaryWrapper
+makeDocumentedCWrapper() {
+    local src=$(makeCWrapper "$@")
+    local docs=$(documentationString "$src")
+    printf "%s\n" "$src"
```

Comment:
This applies to the whole file: prefer `"\\n"` or `'\n'` (it could be argued that `"\n"` is an invalid escape sequence, but Bash just ignores it). For printf format strings, single quotes are usually nicer.

Reply:
So if I understand correctly, I should change
```bash
printf "%s\n" "$src"
```

to
```bash
printf '%s\n' "$src"
```

across the entire file?

Reply:
Yes